### PR TITLE
Reviews from the metric collection and api pr.

### PR DESCRIPTION
**Pull request by ph**

### CODE REVIEW

File: logstash-core/lib/logstash/agent.rb [hunk 1]
```diff
@@ -18,7 +18,9 @@
 LogStash::Environment.load_locale!
 
 class LogStash::Agent
-  attr_reader :metric, :debug, :node_name, :started_at, :pipelines, :logger
+  STARTED_AT = Time.now
```
Comment (purbon) [inline logstash-core/lib/logstash/agent.rb]: you should freeze the constant.

File: logstash-core/lib/logstash/api/lib/app/command.rb [hunk 2]
```diff
@@ -23,7 +23,7 @@ def uptime
     end
 
     def started_at
-      (service.agent.started_at.to_f*1000.0).to_i
+      (LogStash::Agent::STARTED_AT.to_f * 1000.0).to_i
```
Comment (purbon) [inline logstash-core/lib/logstash/api/lib/app/command.rb]: should you always store the .to_f?

**Comment by purbon** [comment]
left only small comment, in general LGTM.

File: logstash-core/lib/logstash/agent.rb (hunk 1 above)
Comment (ph) [inline logstash-core/lib/logstash/agent.rb]: +1

File: logstash-core/lib/logstash/api/lib/app/command.rb (hunk 2 above)
Comment (ph) [inline logstash-core/lib/logstash/api/lib/app/command.rb]: I don't think its needed, its not hit really often. I don't believe we need this optimization

**Pull request closed without merging**